google-ads-migration-examples/src/main/java/com/google/ads/googleads/migration/createcampaign/CreateCompleteCampaignGoogleAdsApiOnly.java

```diff
@@ -0,0 +1,628 @@
+      SearchPagedResponse response = googleAdsServiceClient.search(request);
+
+      // Create and return a list of adGroupCriteria
+      List<AdGroupCriterion> adGroupCriteria = new ArrayList<>();
+      for (GoogleAdsRow googleAdsRow : response.iterateAll()) {
+        adGroupCriteria.add(googleAdsRow.getAdGroupCriterion());
+      }
+      return adGroupCriteria;
+    }
+  }
+
+  /**
+   * Creates keywords ad group criteria.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the client customer ID.
+   * @param adGroup the ad group for the new criteria.
+   * @param keywordsToAdd the keywords to add to the text ads.
+   * @throws GoogleAdsException if an API request failed with one or more service errors.
+   */
+  private List<AdGroupCriterion> createKeywords(
+      GoogleAdsClient googleAdsClient,
+      long customerId,
+      AdGroup adGroup,
+      List<String> keywordsToAdd) {
+    String adGroupResourceName = ResourceNames.adGroup(customerId, adGroup.getId().getValue());
+
+    List<AdGroupCriterionOperation> operations = new ArrayList<>();
+
+    for (String keywordText : keywordsToAdd) {
+      // Create the keyword criterion
+      AdGroupCriterion adGroupCriterion =
+          AdGroupCriterion.newBuilder()
+              .setAdGroup(StringValue.of(adGroupResourceName))
+              .setStatus(AdGroupCriterionStatus.ENABLED)
+              .setKeyword(
+                  KeywordInfo.newBuilder()
+                      .setText(StringValue.of(keywordText))
+                      .setMatchType(KeywordMatchType.EXACT)
+                      .build())
+              .build();
+
+      // Create the operation.
+      AdGroupCriterionOperation op =
+          AdGroupCriterionOperation.newBuilder().setCreate(adGroupCriterion).build();
+      operations.add(op);
+    }
+
+    // Get the AdGroupCriterionService.
+    try (AdGroupCriterionServiceClient adGroupCriterionServiceClient =
+        googleAdsClient.getLatestVersion().createAdGroupCriterionServiceClient()) {
+      // Add the keywords
+      MutateAdGroupCriteriaResponse response =
+          adGroupCriterionServiceClient.mutateAdGroupCriteria(
+              Long.toString(customerId), operations);
+      System.out.printf("Added %d keywords:%n", response.getResultsCount());
+      // Create a list of new keyword resource names
+      List<String> newCriteriaResourceNames = new ArrayList<>();
+      for (MutateAdGroupCriterionResult result : response.getResultsList()) {
+        newCriteriaResourceNames.add(result.getResourceName());
+      }
+
+      // Retrieve the newly created keywords.
+      List<AdGroupCriterion> newCriteria =
+          getKeywords(googleAdsClient, customerId, newCriteriaResourceNames);
+      // Display the results.
+      for (AdGroupCriterion newCriterion : newCriteria) {
+        System.out.printf(
+            "Keyword with text '%s', id %s, and match type '%s' was retrieved for ad group '%s'.%n",
```

Comment:
nit: `id` => `ID` for consistency with existing examples

Reply:
Done.